### ISSUE TITLE
Links to Welsh versions of content should be understandable even when out of context

### ISSUE THREAD

**Issue by CharlotteDowns** [alphagov/govuk-design-system#2796]
This issue is from May 2023 external accessibility audit report.

**WCAG Reference:**
2.4.9 Link Purpose (Link Only) (Level AAA)
[Understanding Link Purpose (Link Only) ](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-link-only.html)| [How to Meet Link Purpose (Link Only)](https://www.w3.org/WAI/WCAG21/quickref/#link-purpose-link-only)

**Issue ID:**
DAC_Link_Purpose_Out_Of_Context_01

**URLs:**
[https://signin.account.gov.uk/sign-in-or-create](https://signin.account.gov.uk/sign-in-or-create)

## Screen shots
![image](https://github.com/alphagov/govuk-frontend/assets/56260216/a85f1190-ae33-4287-be3a-d4e0d3860a1f)

The link provided for users to change the language of the page to Welsh, has been included within a paragraph to provide that context. However, for users navigating the page out of context using the link text alone, the links purpose is not made entirely clear, as the link text is only presented to the user as ‘In Welsh (Cymraeg)’.

## Current code ref(s)
`#main-content > div > div > div`

## Screen-reader comments
> “When I moved out of context for example by using the JAWS links dialogue menu ‘Insert + f7’ I located a link which began with the phrase ‘In Welsh’. It was not immediately clear to me from the link text what would be displayed in Welsh. 

> When I moved in context with the down arrow key for example, I found that a piece of text advising me that the service could be displayed in Welsh appeared directly before the link but had not been included within the link text itself. 

> Including this text within the link text will enable me to make an informed selection both in and out of context.”

## Solution
Ensure that links are descriptive enough when navigating out of context, allowing for users of screen reading assistive technologies to discern their purpose and functionality when doing so.
The link should include more text to clearly inform the user that it is used to change the language to Welsh.

**Comment by gtvj** [comment]
We have merged the above PR today that includes the link purpose within the link text. Link text is now the English and Welsh equivalents of "use GOV.UK One Login in Welsh (Cymraeg)"

**Comment by davidc-gds** [comment]
Hey @claireashworth, since we don't currently have a 'translation toggle link' component or pattern, I'm thinking guidance around how to word translation toggle links could sit in the ['links' portion of our typography style guidance](https://design-system.service.gov.uk/styles/typography/#links)?

It seems like a convention we should standardise, since it is a fairly important piece to get right from a usability and accessibility standpoint.

As an example, perhaps the guidance could say something like:
> `<h3>` Translation toggle links `</h3>`
> When a link is being used to guide people to different translations of a page or service's content, be sure that the link text contains enough context to be understandable on its own.
> For example, the link can say "use [name of service] in Welsh (Cymraeg)" instead of just "Welsh (Cymraeg)".

Thoughts on this? It seems there's a few conversations to be had:
- What standard translation toggle link text should say
- Where the guidance should be placed, and in which format
- What the surrounding guidance content should include

**Comment by CharlotteDowns** [comment]
Need to share this finding with Head of Content, GOV.UK on Monday 21st August

**Comment by claireashworth** [comment]
I seem to recall having this conversation, but obviously did not record it here. We can probably reference the content communities guidance on publishing in Welsh as there's a rigorous process for translation and making sure that translation is findable. For example, one of the steps for adding Welsh content is:

3/ Add the standard call out
Add the following call outs so that users know the content is also available in English or Welsh.
For the Welsh version use:

^Mae'r canllaw hwn hefyd ar gael [yn Saesneg (English)](/link-to-English-equivalent).^

For the English version use:

^This guide is also available [in Welsh (Cymraeg)](/link-to-Welsh-equivalent).^

For start pages use:

^Mae’r dudalen hon ar gael hefyd [yn Saesneg (English)](/link-to-English-equivalent).^

and

^This service is also available [in Welsh (Cymraeg)](/link-to-Welsh-equivalent).^

Place the callout near the top of the first chapter. Most call outs are placed as the second paragraph so that users have a consistent way of knowing a different language is available. But you can use your judgement to decide exactly where to put it. If the callout gets in the way of the main call to action, you can put it further down the page.

I'll also share a presentation showing research on translations on gov.uk.

**Comment by CharlotteDowns** [comment]
This issue can be resolved in the short term with the Links to meet WCAG 2.1 pull request #3044, which as guidance to the links sections of the Typography guidance.

The longer term, more comprehensive resolution would be to have more in-depth guidance, and potentially a new pattern or component for translations. 

There seems to be a lot to take into consideration when expanding the link to increase the amount of text needed to be translated. More detail on [Translations on GOV.UK](https://docs.google.com/presentation/d/12umYUuFI7gxAf03496fvnqFFPZuNBFDJPVD88-Nt1hY/edit#slide=id.g54e4612e82_0_141) and the surrounding complexities have been explored in this slide deck.
